src/functions/add_book.rb

```diff
@@ -0,0 +1,13 @@
+require_relative '../book'
+
+def add_book
+  print 'Publisher: '
+  publisher = gets.chomp
+  print 'Cover State (good, bad): '
+  cover_state = gets.chomp
+  print 'Publish Date (YYYY-MM-DD): '
+  publish_date = gets.chomp
```

Comment:
I kindly suggest that you do some validation to make sure that a correct cover_state and publish date is passed. Because these values are important in the can_be_archived? method.